sdk/compiler/daml-extension/src/extension.ts

```diff
@@ -184,7 +119,162 @@ export async function activate(context: vscode.ExtensionContext) {
+    for (const entry of json) {
+      manifest[entry["multi-package-directory"]] = entry.environment;
+    }
+    return manifest;
+  } catch (e) {
+    outputChannel.appendLine("Failed to parse ide-manifest file: " + e);
+    return null;
+  }
+}
+
+async function fileExists(path: string) {
+  return new Promise(r => fs.access(path, fs.constants.F_OK, e => r(!e)));
+}
+
+async function showRecommendedDirenvPage() {
+  await vscode.commands.executeCommand("extension.open", "mkhl.direnv");
+  vscode.window.showInformationMessage(
+    "After installing, reload the VSCode window or restart the extension host through the command palette.",
+  );
+}
+
+async function startLanguageServers(context: ExtensionContext) {
+  const config = vscode.workspace.getConfiguration("daml");
+  const consent = await getTelemetryConsent(config, context);
+  const rootPath = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
+  if (!rootPath) throw "Couldn't find workspace root";
+  const multiIDESupport = config.get("multiPackageIdeSupport");
+  const gradleSupport = config.get("multiPackageIdeGradleSupport");
+
+  var isGradleProject = false;
+  const gradlewExists = await fileExists(rootPath + "/gradlew");
+  // Of the 3 direnv extensions I found, mkhl.direnv is most popular and only one that populates environment variables for other extensions
+  // As such, it is the one we recommend.
+  const envrcExistsWithoutExt =
+    (await fileExists(rootPath + "/.envrc")) &&
+    vscode.extensions.getExtension("mkhl.direnv") == null;
+
+  if (envrcExistsWithoutExt && gradleSupport) {
+    const warningMessage =
+      "Found an .envrc file but the recommended direnv VSCode extension was not installed. Daml IDE may fail to start due to missing environment." +
+      "\nWould you like to install this extension or attempt to continue without it?";
+    const installAnswer = "Install extension";
+    const doNotInstallAnswer = "Continue without";
+    const neverInstallAnswer = "Do not ask again";
+    const doNotInstallkey = "no-install-direnv";
+
+    // Don't ask if we previously selected doNotInstallAnswer
+    if (!context.workspaceState.get(doNotInstallkey, false)) {
+      const selection = await vscode.window.showWarningMessage(
+        warningMessage,
+        installAnswer,
+        doNotInstallAnswer,
+        neverInstallAnswer,
+      );
+
+      if (selection == neverInstallAnswer) {
+        context.workspaceState.update(doNotInstallkey, true);
+        vscode.window.showInformationMessage(
+          "Warning disabled. If you need the extension in future, run Install Daml Recommended Direnv in the command pallette.",
+        );
+      } else if (selection == installAnswer) {
+        await showRecommendedDirenvPage();
+        return;
+      }
+    }
+  }
+
+  if (multiIDESupport && gradleSupport && gradlewExists) {
+    const generateIdeManifestCommand =
+      rootPath + '/gradlew run --args="--project-dir ' + rootPath + '"';
```

Comment:
Make this more specific to the gradle integration.
Perhaps some kind of name? (for future PR) Or run the rule directly, which should be possible

Reply:
I think we're going to run the rule directly, but that requires more work from Carl, who is busy right now. Given this is behind an experimental flag, I will leave it as is for now, as there isn't much I can do without Carl.